core/rio/rdfxml/src/test/java/org/eclipse/rdf4j/rio/rdfxml/RDFXMLPrettyWriterTest.java

```diff
@@ -104,6 +109,58 @@ public void outOfSequenceItemsAreNotAbbreviated() throws RDFHandlerException, IO
 		assertEquals(Arrays.asList("<rdf:RDF", "<rdf:Bag", "<rdf:_0", "<rdf:_2"), rdfLines);
 	}
 
+	@Test
+	public void compactXMLPrintTest() throws RDFHandlerException, IOException, ClassNotFoundException,
+			NoSuchMethodException, IllegalAccessException, InvocationTargetException, InstantiationException {
+		OutputStream outputStream = new ByteArrayOutputStream();
+		PrintStream printStream = new PrintStream(outputStream);
+		System.setOut(printStream);
+
+		RDFWriter rdfXmlWriter = new RDFXMLPrettyWriter(System.out);
+
+		Class<?> arrangedWriterCls = Class.forName("org.eclipse.rdf4j.rio.turtle.ArrangedWriter");
+		Constructor<?> constructor = arrangedWriterCls.getConstructor(RDFWriter.class);
+		constructor.setAccessible(true);
+		RDFWriter arrangedXmlWriter = (RDFWriter) constructor.newInstance(rdfXmlWriter);
+
+		arrangedXmlWriter.set(BasicWriterSettings.PRETTY_PRINT, false);
+		arrangedXmlWriter.set(BasicWriterSettings.INLINE_BLANK_NODES, true);
+		arrangedXmlWriter.set(XMLWriterSettings.COMPACT_XML, true);
```

Comment:
If you get rid of using the ArrangedWriter, you can also remove the first two settings, and just set the third one (but directly on the rdfXmlWriter, of course).